### PR TITLE
Add graph.items()

### PR DESCRIPTION
**Pull request by pokey**
This PR adds support for iterating over all bound components of the graph, along with the key that they are bound to.

### CODE REVIEW

File: microcosm/caching.py [hunk 1]
```diff
@@ -7,36 +7,20 @@
...
-        pass
-
-    @abstractmethod
-    def __setitem__(self, name, component):
-        pass
-
-    @abstractmethod
-    def __delitem__(self, name):
-        pass
-
```
Comment (pokey) [inline microcosm/caching.py]: These are all defined by `collections.abc.MutableMapping`

File: microcosm/caching.py [hunk 2]
```diff
@@ -76,9 +60,6 @@ def __init__(self, scope="default"):
     def __contains__(self, name):
         return name in ProcessCache.CACHES[self.scope]
 
-    def get(self, name):
-        return self.__getitem__(name)
-
```
Comment (pokey) [inline microcosm/caching.py]: This is automatically defined by `MutableMapping`

File: microcosm/caching.py [hunk 3]
```diff
@@ -88,6 +69,12 @@ def __setitem__(self, name, component):
     def __delitem__(self, name):
         del ProcessCache.CACHES[self.scope][name]
 
+    def __iter__(self):
+        return iter(ProcessCache.CACHES[self.scope])
+
+    def __len__(self):
+        return len(ProcessCache.CACHES[self.scope])
+
```
Comment (pokey) [inline microcosm/caching.py]: Need these to fully support `MutableMapping`

File: microcosm/object_graph.py [hunk 4]
```diff
@@ -137,6 +138,12 @@ def _resolve_key(self, key):
 
         return self.assign(key, component)
 
+    def items(self) -> Iterable[Tuple[str, Any]]:
+        """
+        Iterates over tuples of (name, component) for all bound components.
+        """
+        yield from self._cache.items()
+
```
Comment (pokey) [inline microcosm/object_graph.py]: This is the main purpose of this PR

File: microcosm/caching.py (hunk 1 above)
Comment (cdm-ium) [reply microcosm/caching.py]: I'm guessing the definition in the base class will work for our `Cache` class?

Comment (pokey) [reply microcosm/caching.py]: Yeah these are mostly all abstract in the base class and the ones that are not function as expected

**Pull request merged**